internal/dataplane/parser/translators/ingress.go

```diff
@@ -387,3 +393,97 @@ func MaybePrependRegexPrefixForIngressV1Fn(ingress *netv1.Ingress, applyLegacyHe
 		return lo.ToPtr(MaybePrependRegexPrefix(path, regexPrefix, applyLegacyHeuristic))
 	}
 }
+
+type runeType int
+
+const (
+	runeTypeEscape runeType = iota
+	runeTypeMark
+	runeTypeDigit
+	runeTypePlain
+)
+
+func generateRewriteURIConfig(path string) (string, error) {
+	var out []rune
+	in := []rune(path)
+	lastRuneType := runeTypePlain
+	for i := 0; i < len(in); i++ {
```

Comment:
```suggestion
	for _, char := range path {
```
you can iterate like that to have runs, [see](https://go.dev/doc/effective_go#for)